src/xref2/errors.ml

```diff
@@ -103,7 +103,8 @@ module Tools_error = struct
     | simple_module_type_expr_of_module_error
     | simple_module_lookup_error
     | signature_of_module_error
-    | parent_lookup_error ]
+    | parent_lookup_error
+    | reference_lookup_error ]
```

Comment:
This diff is not necessary.
Reference errors are wrapped in "parent errors" because references are kind of an other language, errors don't mix.

Revert:
```diff
diff --git a/src/xref2/errors.ml b/src/xref2/errors.ml
index afba115b7..55469d03b 100644
--- a/src/xref2/errors.ml
+++ b/src/xref2/errors.ml
@@ -103,8 +103,7 @@ module Tools_error = struct
     | simple_module_type_expr_of_module_error
     | simple_module_lookup_error
     | signature_of_module_error
-    | parent_lookup_error
-    | reference_lookup_error ]
+    | parent_lookup_error ]
 
   let pp_reference_kind fmt k =
     let k =
@@ -166,7 +165,9 @@ module Tools_error = struct
     | `Parent_module e -> Format.fprintf fmt "Parent_module: %a" pp (e :> any)
     | `Fragment_root -> Format.fprintf fmt "Fragment root"
     | `Parent_type e -> Format.fprintf fmt "Parent_type: %a" pp (e :> any)
-    | `Reference e -> pp fmt (e :> any)
+    | `Reference e -> pp_reference_lookup_error fmt e
+
+  and pp_reference_lookup_error fmt = function
     | `Wrong_kind (expected, got) ->
         let pp_sep fmt () = Format.fprintf fmt " or " in
         Format.fprintf fmt "is of kind %a but expected %a" pp_reference_kind got
@@ -178,6 +179,7 @@ module Tools_error = struct
         | #reference_kind as kind ->
             Format.fprintf fmt "Couldn't find %a %S" pp_reference_kind kind name
         )
+    | `Parent e -> pp fmt (e :> any)
 end
 
 (* Ugh. we need to determine whether this was down to an unexpanded module type error. This is horrendous. *)
@@ -210,9 +212,6 @@ let is_unexpanded_module_type_of =
     | `OpaqueClass -> false
     | `Reference (`Parent p) -> inner (p :> any)
     | `Reference _ -> false
-    | `Wrong_kind _ -> false
-    | `Lookup_by_name _ -> false
-    | `Find_by_name _ -> false
   in
   inner
 
@@ -246,7 +245,6 @@ let rec kind_of_error : Tools_error.any -> kind option = function
   | `Parent (`Parent_expr e) -> kind_of_error (e :> Tools_error.any)
   | `Parent (`Parent_module e) -> kind_of_error (e :> Tools_error.any)
   | `Parent (`Parent _ as e) -> kind_of_error (e :> Tools_error.any)
-  | `Reference e -> kind_of_error (e :> Tools_error.any)
   | `OpaqueModule ->
       (* Don't turn OpaqueModule warnings into errors *)
       Some `OpaqueModule
```

Reply:
Why on earth are reference errors 'parent' errors? That doesn't make sense either.

Reply:
That's how the errors are constructed and destructed currently. I have a work-in-progress patch that entirely remove the "parent" thing but it's far from ready (https://github.com/Julow/odoc/tree/imp-tools-errors)

Reply:
ugh. OK, let's leave it as-is for now and sort it out in 2.1